modules/ardrivo/src/Servo.cxx

```diff
@@ -0,0 +1,78 @@
+/*
+ *  Servo.cxx
+ *  Copyright 2020 ItJustWorksTM
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+#include "Arduino.h"
+#include "BoardDataDef.hxx"
+#include "Servo.h"
+
+constexpr int servo_min(int8_t min) { return MIN_PULSE_WIDTH - min * 4; }
+constexpr int servo_max(int8_t max) { return MAX_PULSE_WIDTH - max * 4; }
+constexpr uint8_t TRIM_DURATION = 2;
+
+Servo::Servo() {
+    is_attached = false;
+}
+
+uint8_t Servo::attach(int pin) { return this->attach(pin, MIN_PULSE_WIDTH, MAX_PULSE_WIDTH); }
+
+uint8_t Servo::attach(int pin, int min, int max) {
+    if (board_info->pins_caps[pin].pwm_able == 1) {
+        attached_pin = pin;
+        this->min = (MIN_PULSE_WIDTH - min) / 4;
+        this->max = (MAX_PULSE_WIDTH - max) / 4;
+        this->is_attached = true;
+        return this->attached_pin;
+    }
+}
+
+void Servo::detach() { this->is_attached = false; }
+
+void Servo::write(int value) {
+    if (attached()) {
+        value = std::clamp(value, 0, 180);
+        board_data->servo_value[this->attached_pin] = value;
```

Comment:
You can make this a one liner, no need to set value first.
`board_data->servo_value[this->attached_pin] = std::clamp(value, 0, 180);`